presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/ResourceEstimatesRange.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups;
+
+import com.facebook.presto.spi.session.ResourceEstimates;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public final class ResourceEstimatesRange
+{
+    private final Optional<Range<Duration>> executionTimeRange;
+    private final Optional<Range<Duration>> cpuTimeRange;
+    private final Optional<Range<DataSize>> peakMemoryRange;
+
+    @JsonCreator
+    public ResourceEstimatesRange(
+            @JsonProperty("executionTimeRange") Optional<Range<Duration>> executionTimeRange,
+            @JsonProperty("cpuTimeRange") Optional<Range<Duration>> cpuTimeRange,
+            @JsonProperty("peakMemoryRange") Optional<Range<DataSize>> peakMemoryRange)
+    {
+        this.executionTimeRange = requireNonNull(executionTimeRange, "executionTimeRange is null");
+        this.cpuTimeRange = requireNonNull(cpuTimeRange, "cpuTimeRange is null");
+        this.peakMemoryRange = requireNonNull(peakMemoryRange, "peakMemoryRange is null");
+    }
+
+    @JsonProperty
+    public Optional<Range<Duration>> getExecutionTimeRange()
+    {
+        return executionTimeRange;
+    }
+
+    @JsonProperty
+    public Optional<Range<Duration>> getCpuTimeRange()
+    {
+        return cpuTimeRange;
+    }
+
+    @JsonProperty
+    public Optional<Range<DataSize>> getPeakMemoryRange()
+    {
+        return peakMemoryRange;
+    }
+
+    boolean match(ResourceEstimates resourceEstimates)
+    {
+        if (executionTimeRange.isPresent() &&
```

Comment:
This set of conditional statements is quite confusing - is there a way to simplify this so it better matches what one would expect? If not, that's OK, just a suggestion.

Reply:
@raghavsethi : I feel part of the complication comes from both of them are optional. Try to massage it a bit (make the optional estimate an local variable) and hope it's better to read .